tv_apps/browser/js/sync/tabNavigation.js

```diff
@@ -0,0 +1,139 @@
+/* global LazyLoader */
+/* global Awesomescreen */
+/* global SyncBrowserDB */
+/* global SyncManagerBridge */
+/* global SettingsListener */
+/* global FirefoxSyncTabList */
+
+'use strict';
+
+(function(exports) {
+  const DEBUG = false;
+  function debug() {
+    if (DEBUG) {
+      console.log('[FirefoxSyncTabNavigation] ' +
+        Array.slice(arguments).concat());
+    }
+  }
+
+  const TAB_SETTING = 'sync.collections.tabs.enabled';
+
+  var FirefoxSyncTabNavigation = {
+    init() {
+      this.isTabOptionEnabled = false;
+
+      this.tabOptionEl = document.getElementById('sync-tab-option');
+
+      LazyLoader.load('shared/js/settings_listener.js').then(() => {
+        SettingsListener.observe(TAB_SETTING, true, enabled => {
+          debug(TAB_SETTING + ' changed to ' + enabled);
+          this.isTabOptionEnabled = enabled;
+          if (enabled) {
+            this.showTabNavigationOption();
+            this.tabOptionEl.addEventListener('mousedown',
+              this.handleTabOptionMouseDown.bind(this));
+          } else {
+            this.hideTabNavigationOption();
+            this.tabOptionEl.removeEventListener('mousedown',
+              this.handleTabOptionMouseDown.bind(this));
+            this.restoreTabNavigation();
+          }
+        });
+      });
+
+      SyncManagerBridge.addListener(this.onSyncChange.bind(this));
+
+      document.addEventListener('visibilitychange', () => {
+        if (!document.hidden) {
+          this.refresh();
+        }
+      });
+
+      this.refresh();
+    },
+
+    refresh() {
+      SyncManagerBridge.getInfo().then(this.onSyncChange.bind(this));
+    },
+
+    onSyncChange(message) {
+      debug(JSON.stringify(message));
+      if (!message || !message.state) {
+        throw new Error('Missing sync state');
+      }
+      switch (message.state) {
+        case 'disabled':
+          if (this.isTabNavigationOptionDisplayed()) {
+            FirefoxSyncTabList.clean();
+            this.restoreTabNavigation();
+            this.hideTabNavigationOption();
+          }
+          break;
+        case 'enabled':
+          /**
+           * Show sync tab option when sync tab collection is enables but the
+           * sync tab option didn't display
+           */
+          if (this.isTabOptionEnabled &&
+            !this.isTabNavigationOptionDisplayed()) {
+            this.showTabNavigationOption();
+          }
+
+          /**
+           * Update FirefoxSyncTabList when sync-tab option is enable
+           */
+          if (this.isTabNavigationOptionDisplayed()) {
+            SyncBrowserDB.getAllDeviceTabs(tabData => {
+              FirefoxSyncTabList.update(tabData);
+            });
+          }
+          break;
+      }
+      this.state = message.state;
+    },
+
+    showTabNavigationOption() {
+      this.tabOptionEl.hidden = false;
```

Comment:
According to https://html.spec.whatwg.org/multipage/interaction.html#the-hidden-attribute , the ```hidden``` attribute is not intended to be use in this kind of condition. We better use CSS style to hide the element instead.